File: 003_backend/web-api/Models/DetailModels/SchoolYearDetails.cs
using System;
namespace web_api.Models.DetailModels
{
    public class SchoolYearDetails
    {
        public Guid Id { get; set; }
        public string? Name { get; set; }

        public SchoolDetails? School { get; set; }
        public IList<SubjectDetails>? Subjects { get; set; } = new List<SubjectDetails>();
    }
}

File: 002_frontend/src/app/api/models/school-year-details.ts
/* tslint:disable */
/* eslint-disable */
import { SchoolDetails } from './school-details';
import { SubjectDetails } from './subject-details';
export interface SchoolYearDetails {
  id?: string;
  name?: null | string;
  school?: SchoolDetails;
  subjects?: null | Array<SubjectDetails>;
}
